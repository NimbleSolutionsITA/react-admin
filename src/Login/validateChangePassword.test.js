import validate from './validateChangePassword';

const translate = () => 'Required';

test('should validate that the password exists', () =>
  expect(validate({}, { translate })).toMatchSnapshot());

test('should validate that the password has a minimum length', () =>
  expect(validate({ password: '123' }, { translate })).toMatchSnapshot());

test('should validate that the password has uppercase characters', () =>
  expect(validate({ password: 'abcabcabc' }, { translate })).toMatchSnapshot());

test('should validate that the password has lowercase characters', () =>
  expect(validate({ password: 'ABCABCABC' }, { translate })).toMatchSnapshot());

test('should validate that the password has a number', () =>
  expect(validate({ password: 'abcABCabc' }, { translate })).toMatchSnapshot());

test('should validate that the password has a special character', () =>
  expect(validate({ password: 'abcABC123' }, { translate })).toMatchSnapshot());

test('should have no error messages if the password is valid', () =>
  expect(validate({ password: 'abc|ABC12' }, { translate })).toEqual({}));
