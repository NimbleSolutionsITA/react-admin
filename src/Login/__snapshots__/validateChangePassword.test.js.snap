// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should validate that the password exists 1`] = `
Object {
  "password": "Required",
}
`;

exports[`should validate that the password has a minimum length 1`] = `
Object {
  "password": "Password must be at least 8 characters.",
}
`;

exports[`should validate that the password has a number 1`] = `
Object {
  "password": "Password must contain a number.",
}
`;

exports[`should validate that the password has a special character 1`] = `
Object {
  "password": "Password must contain a special character.",
}
`;

exports[`should validate that the password has lowercase characters 1`] = `
Object {
  "password": "Password must contain an lowercase character.",
}
`;

exports[`should validate that the password has uppercase characters 1`] = `
Object {
  "password": "Password must contain an uppercase character.",
}
`;
